token/services/db/sql/sqlite/tokenlock.go

```diff
@@ -7,40 +7,70 @@ SPDX-License-Identifier: Apache-2.0
 package sqlite
 
 import (
-	"fmt"
 	"time"
 
-	common2 "github.com/hyperledger-labs/fabric-smart-client/platform/view/services/db/driver/common"
+	common3 "github.com/hyperledger-labs/fabric-smart-client/platform/view/services/db/driver/common"
+	q "github.com/hyperledger-labs/fabric-smart-client/platform/view/services/db/driver/sql/query"
+	common2 "github.com/hyperledger-labs/fabric-smart-client/platform/view/services/db/driver/sql/query/common"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/db/driver/sql/sqlite"
 	"github.com/hyperledger-labs/fabric-token-sdk/token/services/db/driver"
 	"github.com/hyperledger-labs/fabric-token-sdk/token/services/db/sql/common"
 )
 
 type TokenLockStore struct {
 	*common.TokenLockStore
+	ci common2.CondInterpreter
+}
+
+func IsStale(tokenLocks common2.TableName, requests common2.TableName, leaseExpiry time.Duration) *isStale {
+	return &isStale{tokenLocks: tokenLocks, requests: requests, leaseExpiry: leaseExpiry}
+}
+
+type isStale struct {
+	tokenLocks  common2.TableName
+	requests    common2.TableName
+	leaseExpiry time.Duration
+}
+
+func (c *isStale) WriteString(_ common2.CondInterpreter, sb common2.Builder) {
+	sb.WriteString("tx_id IN (SELECT ").
+		WriteString(string(c.tokenLocks)).
+		WriteString(".tx_id FROM ").
+		WriteString(string(c.tokenLocks)).
+		WriteString(" JOIN ").
+		WriteString(string(c.requests)).
+		WriteString(" ON ").
+		WriteString(string(c.tokenLocks)).
+		WriteString(".tx_id = ").
+		WriteString(string(c.requests)).
+		WriteString(".tx_id WHERE ").
+		WriteString(string(c.requests)).
```

Comment:
```
condition := cond.Or(
 cond.In(c.requests.Field("status"), Deleted),
 cond.Before(c.tokenLocks.Field("created_at"), c.leaseExpiry),
)
```
You will need to write `Before`. I can help you if you want.
and then
`WriteConditionSerializable(condition, ci)`